token/token.go

```diff
@@ -41,6 +45,38 @@ const (
 	keywordEnd
 )
 
+// types defines the string representation of token types
+var types = [...]string{
+	Invalid: "Invalid",
+
+	EOF:     "EOF",
+	Comment: "Comment",
+
+	Ident: "Ident",
+	Int:   "Int",
+	Float: "Float",
+
+	Assign:    "=",
+	Semicolon: ";",
+	Colon:     ":",
+
+	Add: "+",
+	Sub: "-",
+	Div: "/",
+	Mod: "%",
+
+	Let: "let",
+}
+
+// String converts a token type to it's string representation.
+func (t Type) String() string {
+	if t < Type(len(types)) {
+		return types[t]
+	}
+
+	return types[Invalid]
+}
```

Comment:
Due to the private begin/end elements this will return an empty string on such a "token-type". I think we should check on this and return either `"Invalid"` or `"Not-a-Token"` (or something along those lines) there. Also I'm not sure if we should generalize all non-valid tokens as `"Invalid"`, maybe we should return something like `"InvalidTokenType(<hex>)"` instead? I think this would make more sense for debugging as we could differentiate between these non-valid/unknown and the `"Invalid"` token(s).